app/src/main/java/com/github/studydistractor/sdp/ui/EventScreen.kt

```diff
@@ -0,0 +1,127 @@
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Chat
+import androidx.compose.material3.Button
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import com.github.studydistractor.sdp.event.EventViewModel
+
+@Composable
+fun EventScreen(
+    eventViewModel : EventViewModel,
+    onOpenChatClick: () -> Unit,
+) {
+    fun showFailureToast(context: Context, message: String) {
+        Toast.makeText(context, "Failure: $message", Toast.LENGTH_SHORT)
+            .show()
+    }
+
+    val uiState by eventViewModel.uiState.collectAsState()
+    val context = LocalContext.current
+
+    LazyColumn(
+        modifier = Modifier
+            .fillMaxWidth()
+            .wrapContentHeight(),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Top
+    ) {
+        item {
+            Spacer(modifier = Modifier.height(16.dp))
+
+            Text(
+                text = uiState.event.name,
+                style = MaterialTheme.typography.headlineLarge,
+                modifier = Modifier.testTag("event-name"),
+            )
+
+            Text(
+                text = uiState.event.description,
+                style = MaterialTheme.typography.bodyLarge,
+                modifier = Modifier.testTag("event-description")
+            )
+
+            Spacer(modifier = Modifier.height(32.dp))
+
+            Text(
+                text = eventViewModel.eventMetadataString(),
+                style = MaterialTheme.typography.bodyLarge,
+                modifier = Modifier.testTag("event-metadata")
+            )
+
+            Spacer(modifier = Modifier.height(32.dp))
+
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth(),
+                horizontalArrangement = Arrangement.SpaceEvenly,
+            ) {
+                Button(
+                    onClick = {
+                        eventViewModel.toggleParticipation().addOnFailureListener {
+                            showFailureToast(context, "Failed to update participation !")
+                        }
+                    },
+                    modifier = Modifier.testTag("toggle-participate")
+                ) {
+                    Text(
+                        text = uiState.toggleParticipationButtonText,
+                        color = Color.White
```

Comment:
Please use Material color scheme instead, it is also valid for other component in the same file

Reply:
ye! sorry